tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/multiprotocol/JMSMismatchedRoutingTypeTest.java

```diff
@@ -0,0 +1,112 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.tests.integration.jms.multiprotocol;
+
+import javax.jms.Connection;
+import javax.jms.MessageProducer;
+import javax.jms.Queue;
+import javax.jms.Session;
+import javax.jms.Topic;
+
+import org.apache.activemq.artemis.api.core.QueueConfiguration;
+import org.apache.activemq.artemis.api.core.RoutingType;
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.core.server.impl.AddressInfo;
+import org.apache.activemq.artemis.tests.util.RandomUtil;
+import org.junit.Test;
+
+public class JMSMismatchedRoutingTypeTest extends MultiprotocolJMSClientTestSupport {
+
+   protected final String ANYCAST_ADDRESS = RandomUtil.randomString();
+   protected final String MULTICAST_ADDRESS = RandomUtil.randomString();
+
+   @Override
+   protected boolean isAutoCreateAddresses() {
+      return false;
+   }
+
+   @Override
+   protected boolean isAutoCreateQueues() {
+      return false;
+   }
+
+   @Override
+   protected void createAddressAndQueues(ActiveMQServer server) throws Exception {
+      server.addAddressInfo(new AddressInfo(SimpleString.toSimpleString(ANYCAST_ADDRESS), RoutingType.ANYCAST));
+      server.createQueue(new QueueConfiguration(RandomUtil.randomString()).setAddress(ANYCAST_ADDRESS).setRoutingType(RoutingType.ANYCAST));
+
+      server.addAddressInfo(new AddressInfo(SimpleString.toSimpleString(MULTICAST_ADDRESS), RoutingType.MULTICAST));
+      server.createQueue(new QueueConfiguration(RandomUtil.randomString()).setAddress(MULTICAST_ADDRESS).setRoutingType(RoutingType.MULTICAST));
+   }
+
+   @Test
+   public void testSendingMulticastToAnycastAMQP() throws Exception {
+      internalTestSendingMulticastToAnycast(AMQPConnection);
+   }
+
+   @Test
+   public void testSendingMulticastToAnycastCore() throws Exception {
+      internalTestSendingMulticastToAnycast(CoreConnection);
+   }
+
+   @Test
+   public void testSendingMulticastToAnycastOpenWire() throws Exception {
+      internalTestSendingMulticastToAnycast(OpenWireConnection);
+   }
+
+   private void internalTestSendingMulticastToAnycast(ConnectionSupplier connectionSupplier) throws Exception {
+      Connection connection = connectionSupplier.createConnection();
+      Session s = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+      Topic topic = s.createTopic(ANYCAST_ADDRESS);
+      try {
+         MessageProducer producer = s.createProducer(topic);
+         producer.send(s.createMessage());
+         fail("Sending message here should fail!");
+      } catch (Exception e) {
+         e.printStackTrace();
```

Comment:
Maybe just a simple log message instead of the full stack? Could also debug log with the exception if desired (here and below)? The test output is long enough without another bunch of stacktraces in it for things expected to happen.